From: Petr Mladek <pmladek@suse.cz>
Date: Wed, 30 Jul 2014 14:24:27 +0200
Subject: kgr: confusion with fatal state
Patch-mainline: submitted for review
References: fate#313296

The patch state is currently shown the following way:

    $> cat /sys/kernel/kgraft/sample_patcher/state
    Function              State  Fatal
    SyS_iopl                  1      1
    capable                   1      1
    unknown_function          5      0
    ftr_test_filter_read      1      0

I am always scared when I see that some functions have the fatal set.
It has somehow negative meaning about that something went wrong very badly.
This patch changes the output to:

    $> cat /sys/kernel/kgraft/sample_patcher/state
    Function               Weak  State
    SyS_iopl                  0      1
    capable                   0      1
    unknown_function          1      5
    ftr_test_filter_read      1      1

The meaning is that the symbol is weak. We even might change the
"abort_if_missing" variable to "weak" and revert the condition
everywhere.

Signed-off-by: Petr Mladek <pmladek@suse.cz>
Acked-by: Jiri Kosina <jkosina@suse.cz>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 kernel/kgraft_files.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/kernel/kgraft_files.c b/kernel/kgraft_files.c
index 9bcc1b5ae317..5e2da03d3a1f 100644
--- a/kernel/kgraft_files.c
+++ b/kernel/kgraft_files.c
@@ -46,12 +46,13 @@ static ssize_t state_show(struct kobject *kobj, struct kobj_attribute *attr,
 	const struct kgr_patch_fun *pf;
 	ssize_t size;
 
-	size = snprintf(buf, PAGE_SIZE, "%-20s  State  Fatal\n", "Function");
+	size = snprintf(buf, PAGE_SIZE, "%-20s   Weak  State\n", "Function");
+
 
 	kgr_for_each_patch_fun(p, pf) {
 		size += snprintf(buf + size, PAGE_SIZE - size,
-				"%-20s  %5d  %5d\n",
-				pf->name, pf->state, pf->abort_if_missing);
+				"%-20s  %5d  %5d\n", pf->name,
+				 !(pf->abort_if_missing), pf->state);
 	}
 	return size;
 }
-- 
2.0.4

